katsdptelstate/memory.py

```diff
@@ -112,18 +112,29 @@ def __init__(self, data):
 
     def set(self, key, value, expiry, info):
         self.data[key] = value
-        self.n_keys += 1
+        if not key.startswith(TelescopeState._INTERNAL_MARKER):
+            self.n_keys += 1
+        elif key == TelescopeState._SEPARATOR_KEY:
+            self.separator = value
 
     def start_sorted_set(self, key, length, expiry, info):
         self.data[key] = []
-        self.n_keys += 1
+        if not key.startswith(TelescopeState._INTERNAL_MARKER):
+            self.n_keys += 1
+        elif key == TelescopeState._SEPARATOR_KEY:
+            raise TypeError('Separator must be immutable')  # Will be remapped to RdbParseError
 
     def zadd(self, key, score, member):
         self.data[key].append(member)
 
     def end_sorted_set(self, key):
         self.data[key].sort()
 
+    def end_rdb(self):
```

Comment:
It could if the user tries to do operations on the telstate (whether through the same object or through another connection to redis) while the loading is happening. That's why I was keen to have "create a brand new memory-backend telstate from a file" as the only supported operation, with anything else being for expert use only.